Utils/AsyncTaskList.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DUCK.Utils
+{
+	public class AsyncTaskList
+	{
+		private readonly List<Action<Action>> tasks = new List<Action<Action>>();
+		private int currentTaskIndex;
+
+		/// <summary>
+		/// Adds an async task to the task list. The Action provided takes another action called next.
+		/// Call this next function to indicate the task is complete and the next one in the chain, should be called
+		/// Eg:
+		/// taskList.Add(next => {
+		///   // do some work, then call next when complete.
+		///   next();
+		/// });
+		/// </summary>
+		/// <param name="taskFunction">The task function</param>
+		public void Add(Action<Action> taskFunction)
+		{
+			tasks.Add(taskFunction);
+		}
+
+		/// <summary>
+		/// Adds a synchronous task to the task list.
+		/// This is a simple function that is executed as part of async task list.
+		/// After the function has been called, the next task will begin instantly
+		/// </summary>
+		/// <param name="taskFunction">The task function</param>
+		public void Add(Action taskFunction)
+		{
+			Add(next =>
+			{
+				taskFunction();
+				next();
+			});
+		}
+
+		/// <summary>
+		/// Runs the task list, in order and calls the onComplete callback when all tasks have been completed.
+		/// </summary>
+		/// <param name="onComplete">Optional on complete callback</param>
+		public void Run(Action onComplete = null)
+		{
+			currentTaskIndex = 0;
```

Comment:
Wondering if this has issues if Run is called while an async action is pending?

Reply:
Good point, I'll stick a guard in, for attempting to run whilst in progress